src/Initializers/ConsoleInitializer.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Initializers;
+
+use Tempest\Application;
+use Tempest\Console\Commands\ScheduleTaskCommand;
+use Tempest\Console\Components\GenericComponentRenderer;
+use Tempest\Console\Components\UnsupportedComponentRenderer;
+use Tempest\Console\Console;
+use Tempest\Console\ConsoleArgumentBag;
+use Tempest\Console\Exceptions\ConsoleExceptionHandler;
+use Tempest\Console\GenericConsole;
+use Tempest\Console\Highlight\TempestTerminalTheme;
+use Tempest\Console\Highlight\TextTerminalTheme;
+use Tempest\Console\Input\StdinInputBuffer;
+use Tempest\Console\Input\UnsupportedInputBuffer;
+use Tempest\Console\Output\LogOutputBuffer;
+use Tempest\Console\Output\StdoutOutputBuffer;
+use Tempest\Container\Container;
+use Tempest\Container\Initializer;
+use Tempest\Container\Singleton;
+use Tempest\Highlight\Highlighter;
+
+#[Singleton]
+class ConsoleInitializer implements Initializer
+{
+    public function initialize(Container $container): Console
+    {
+        $argumentBag = $container->get(ConsoleArgumentBag::class);
+
+        if ($argumentBag->getCommandName() === ScheduleTaskCommand::NAME) {
+            return $this->backgroundTaskConsole($container);
+        }
+
+        return $this->cliConsole($container);
+    }
+
+    public function backgroundTaskConsole(Container $container): GenericConsole
+    {
+        $application = $container->get(Application::class);
+        $textHighlighter = new Highlighter(new TextTerminalTheme());
+
+        $container->singleton(ConsoleExceptionHandler::class, fn () => new ConsoleExceptionHandler(
+            console: $container->get(Console::class),
+            application: $application,
+            highlighter: $textHighlighter,
```

Comment:
We also have to register another exception handler with another highlighter, because otherwise it would print formatted code into the log, which we don't want. 

We could move this part to a dedicated initializer if people find that better.